src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -0,0 +1,55 @@
+package com.example.onlinebookstore.controller;
+
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.CreateBookRequestDto;
+import com.example.onlinebookstore.service.BookService;
+import jakarta.validation.Valid;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto findById(@PathVariable Long id) {
```

Comment:
You can also add validation to @PathVariable. If you do this you need to add @Validated annotation on your controller class

```suggestion
    public BookDto findById(@PathVariable @Positive Long id) {
```